include/dd/Edge.hpp

```diff
@@ -37,6 +37,164 @@ namespace dd {
         [[nodiscard]] static constexpr Edge terminal(const Complex& w) { return {Node::terminal, w}; }
         [[nodiscard]] constexpr bool        isZeroTerminal() const { return Node::isTerminal(p) && w == Complex::zero; }
         [[nodiscard]] constexpr bool        isOneTerminal() const { return Node::isTerminal(p) && w == Complex::one; }
+
+        [[nodiscard]] static inline Node* setDensityConjugateTrue(const Node* p) {
+            return reinterpret_cast<Node*>(reinterpret_cast<std::uintptr_t>(p) | 1ULL);
+        }
+
+        [[nodiscard]] static inline Node* setFirstEdgeDensityPathTrue(const Node* p) {
+            return reinterpret_cast<Node*>(reinterpret_cast<std::uintptr_t>(p) | 2ULL);
+        }
+
+        [[nodiscard]] static inline Node* setDensityMatrixTrue(const Node* p) {
+            return reinterpret_cast<Node*>(reinterpret_cast<std::uintptr_t>(p) | 4ULL);
+        }
+
+        //        static inline void setDensityConjugateEdgeTrue(Edge *e) {
+        //            e->p = reinterpret_cast<Node*>(reinterpret_cast<std::uintptr_t>(e->p) | 1ULL);
+        //        }
+        //
+        //        static inline void setDensityMatrixEdgeTrue(Edge *e) {
+        //            e->p = reinterpret_cast<Node*>(reinterpret_cast<std::uintptr_t>(e->p) | 4ULL);
+        //        }
+
+        [[nodiscard]] static inline bool isDensityConjugateSet(const long p) {
+            return p & 1ULL;
+        }
+
+        [[nodiscard]] static inline bool isFirstEdgeDensityPath(const long p) {
+            return p & 2ULL;
+        }
+
+        [[nodiscard]] static inline bool isDensityMatrix(const long p) {
+            return p & 7ULL;
+        }
+
+        [[nodiscard]] static inline bool isDensityMatrixNode(const long p) {
+            return p & 8U;
+        }
```

Comment:
Moved the functions to `Node.hpp`. 
The parameters are of type long, (although the `flags` register is of type `std::uint_least8_t`) since the very same flags are also used within the node pointers